shell/platform/darwin/common/framework/Headers/FlutterBaseDartProject.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERBASEDARTPROJECT_H_
+#define FLUTTER_FLUTTERBASEDARTPROJECT_H_
+
+#import <Foundation/Foundation.h>
+
+#import "FlutterMacros.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A set of Flutter and Dart assets used by a `FlutterEngine` to initialize execution.
+ *
+ * TODO(stuartmorgan): Align API with iOS DartProject, and combine.
+ */
+@interface FlutterBaseDartProject : NSObject
+
+/**
+ * Initializes a Flutter Dart project from a bundle.
+ *
+ * The bundle must either contain a flutter_assets resource directory, or set the Info.plist key
+ * FLTAssetsPath to override that name (if you are doing a custom build using a different name).
+ *
+ * @param bundle The bundle containing the Flutter assets directory. If nil, the App framework
+ *               created by Flutter will be used.
+ */
+- (instancetype)initWithPrecompiledDartBundle:(nullable NSBundle*)bundle NS_DESIGNATED_INITIALIZER;
+
+/**
+ * If set, allows the Flutter project to use the dart:mirrors library.
+ *
+ * Deprecated: This function is temporary and will be removed in a future release.
+ */
+@property(nonatomic) bool enableMirrors;
```

Comment:
iOS doesn't have this property, should we keep this on the macOS embedder's `FlutterDartProject`?